draft-ietf-quic-recovery.md

```diff
@@ -1001,14 +1001,22 @@ packets might cause the sender's bytes in flight to exceed the congestion window
 until an acknowledgement is received that establishes loss or delivery of
 packets.
 
+## Persistent Congestion
+
 When an ACK frame is received that establishes loss of all in-flight packets
-sent prior to a threshold number of consecutive PTOs (pto_count is more than
-kPersistentCongestionThreshold, see {{cc-consts-of-interest}}), the network is
-considered to be experiencing persistent congestion, and the sender's congestion
-window MUST be reduced to the minimum congestion window (kMinimumWindow).  This
-response of collapsing the congestion window on persistent congestion is
-functionally similar to a sender's response on a Retransmission Timeout (RTO) in
-TCP {{RFC5681}}.
+sent over a long enough period of time, the network is considered to be
+experiencing persistent congestion.  Commonly, this can be established by
+consecutive PTOs, but since the PTO timer is reset when a new ack-eliciting
+packet is sent, an explicit duration must be used to account for those cases
+where PTOs do not occur or are substantially delayed.  This duration is the
+equivalent of kPersistentCongestionThreshold consecutive PTOS, smoothed_rtt +
+4 * rttvar + max_ack_delay * ((2 ^ kPersistentCongestionThreshold) - 1).
```

Comment:
Q for here and below: Should max_ack_delay be added after exponential backoff, instead of before?  Adding it after incentivizes implementations to indicate a realistic value, since an unrealistic one just adds to rttvar, which seems desirable.

Reply:
The order is the same in `SetLossDetectionTimer`:
```    // Calculate PTO duration
    timeout =
      smoothed_rtt + 4 * rttvar + max_ack_delay
    timeout = max(timeout, kGranularity)
    timeout = timeout * (2 ^ pto_count)```
Though, on a side issue, I feel the `kGranularity` check should be moved to the end of that calculation...

Reply:
Agreed, what you have matches the current pseudocode.  I'll open an issue for kGranularity and max_ack_delay and when they should be added.

Reply:
Agreed that it seems odd to backoff on max_ack_delay. @ianswett : Let's do that separately.